assets/pages/contributions-thankyou/components/marketingConsentContainer.jsx

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+// ----- Imports ----- //
+
+import { connect } from 'react-redux';
+
+import { setGnmMarketing } from 'helpers/user/userActions';
+import { sendMarketingPreferencesToIdentity } from 'containerisableComponents/marketingConsent/helpers';
+import MarketingConsent from 'containerisableComponents/marketingConsent/marketingConsent';
+
+import type { Csrf as CsrfState } from 'helpers/csrf/csrfReducer';
+
+
+// ----- Component ----- //
+
+
+function mapDispatchToProps(dispatch) {
+  return {
+    onClick: (marketingPreferencesOptIn: boolean, email: string, csfr: CsrfState) => {
+      sendMarketingPreferencesToIdentity(
+        marketingPreferencesOptIn,
+        email,
+        dispatch,
+        csfr,
+        'CONTRIBUTIONS_THANK_YOU',
+      );
+    },
+    marketingPreferenceUpdate: (preference: boolean) => {
+      dispatch(setGnmMarketing(preference));
+    },
```

Comment:
It's possible this can just be `marketingPreferenceUpdate: setGnmMarketing` and Redux will figure out that it needs a dispatch.